src/main/java/factionsystem/Commands/GateCommand.java

```diff
@@ -0,0 +1,86 @@
+package factionsystem.Commands;
+
+import static factionsystem.Subsystems.UtilitySubsystem.createStringFromFirstArgOnwards;
+
+import java.util.Set;
+
+import org.bukkit.ChatColor;
+import org.bukkit.Material;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+import factionsystem.Main;
+import factionsystem.Objects.Faction;
+import factionsystem.Objects.Gate;
+import factionsystem.Subsystems.UtilitySubsystem;
+
+public class GateCommand {
+
+	Main main = null;
+	
+	public GateCommand(Main plugin)
+	{
+		main = plugin;
+	}
+	
+	public void handleGate(CommandSender sender, String[] args)
+	{
+		if (sender instanceof Player)
+		{
+			Player player = (Player) sender;
+			
+			if (player.hasPermission("mf.gate") || player.hasPermission("mf.default"))
+			{
+				if (args.length > 1)
+				{
+					if (args[1].equalsIgnoreCase("cancel"))
+					{
+						if (main.creatingGatePlayers.containsKey(player.getUniqueId()))
+						{
+							main.creatingGatePlayers.remove(player.getUniqueId());
+							player.sendMessage(ChatColor.RED + "Creating gate cancelled!");
+							return;
+						}
+					}
+					if (args[1].equalsIgnoreCase("list"))
+					{
+						Faction faction = UtilitySubsystem.getPlayersFaction(player.getUniqueId(), main.factions);
+						if (faction != null)
```

Comment:
If the faction is null, the player should be told they need to be in a faction to use the command.